docs/source/howto/data.rst

```diff
@@ -248,11 +247,247 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+        filters={
+            'or':[
+                {'attributes.process_state': 'excepted'},
+                {'attributes.process_label': 'ArithmeticAddCalculation'}
+            ]
+        },
+    )
+
+If we had written ``and`` instead of ``or`` in the example above, we would have performed the exact same query as the previous one, because ``and`` is the default behavior if you provide several filters as key-value pairs in a dictionary to the ``filters`` argument.
+In case you want all calculation jobs with state ``finished`` or ``excepted``, you can also use the ``in`` operator:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(
+        CalcJobNode,
+        filters={
+            'attributes.process_state': {'in': ['finished', 'excepted']}
+        },
+    )
+
+You can also negate a filter by adding an exclamation mark in front of the operator.
+So, to query for all calculation jobs that are not a 'finished' or 'excepted' state:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(
+        CalcJobNode,
+        filters={
+            'attributes.process_state': {'!in': ['finished', 'excepted']}
+        },
+    )
+
+.. note::
+
+    The above rule applies to all operators.
+    For example, you can check non-equality with ``!==``, since this is the equality operator (``==``) with a negation prepended.
+
+A list of all implemented operators can be found in the ``QueryBuilder`` docstring.
+
+.. _how-to:data:find:relationships:
+
+Relationships
+-------------
+
+It is possible to query for data based on its relationship to another entity in the database.
+Imagine you are not interested in the calculation jobs themselves, but in one of the outputs they create.
+You can build upon your initial query for all ``CalcJobNode``'s in the database using the relationship of the output to the first step in the query:
+
+.. code-block::
+
+    qb = QueryBuilder()
+    qb.append(CalcJobNode, tag='calcjob')
+    qb.append(Int, with_incoming='calcjob')
+
+In the first ``append`` call, we query for all ``CalcJobNode``'s in the database, and *tag* this step with the *unique* identifier ``'calcjob'``.
+Next, we look for all ``Int`` nodes that are an output of the ``CalcJobNode``'s found in the first step, using the ``with_incoming`` relationship argument.
+Since the ``CalcJobNode`` *created* the ``Int`` node, it is an *incoming* node from the ``Int`` node's perspective.
+
+In the context of our query, we are building a *path* consisting of *vertices* (i.e. the entities we query for) connected by *edges* defined by the relationships between them.
+A list of all the relationships you can use for your query, as well as the entities that they connect, can be found in the ``QueryBuilder`` docstring.
+
+.. _how-to:data:find:projections:
+
+Projections
+-----------
+
+When no *projection* is specified, the default behavior of the ``QueryBuilder`` is to project the entities corresponding to the final vertex of the query path.
```

Comment:
This sentence is a bit complicated and potentially confusing as it mixes two parts of the behavior.
By default, if no explicit projections are specified, the builder will return:
1. entities that were appended last
2. actual instances of the entities

You first sentence refers to the first, but the "however" in your following sentence applies to the second point.
I think the solution is to move the idea of specific property projections to a later point and first explain how to project an entity that was not appended last. You show `project='*'` which then leads nicely into the next section that `*` projects the entire instance, but you can also project just certain properties. Note that I don't think we should call it `attributes` because `label`, `pk`, etc. can also be projected but are not attbributes.